### PR TITLE
feat(financial-aid): Staff can comment on application

### PR DESCRIPTION
**Pull request by MargretFinnboga**
# Staff can comment on application

https://app.asana.com/0/1200826552987384/1200876741986739

## What

Staff can comment on application. State was changed to event type and amount, and staff comment added.
Added to navigation user 

## Why

Required from customer 
Better user experience 

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

### CODE REVIEW

**Review by Rikkiabb — CHANGES_REQUESTED** [review]

File: apps/financial-aid/backend/migrations/20210830110934-update-application_events.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
+'use strict'
+
+module.exports = {
+  up: (queryInterface, Sequelize) => {
+    return queryInterface.sequelize.transaction((t) =>
+      Promise.all([
+        queryInterface.addColumn(
+          'application_events',
+          'amount',
```
[the remark this answers is not in the document]
Comment (MargretFinnboga) [reply apps/financial-aid/backend/migrations/20210830110934-update-application_events.js:9]: new table, id, created. application_id, comment. event_type, amount, staff_comment

[the remark this answers is not in the document]
Comment (MargretFinnboga) [reply apps/financial-aid/backend/migrations/20210830110934-update-application_events.js:9]: Lets changed it up, to id, created, application_id, event_type and comment

File: apps/financial-aid/web-veita/src/components/CommentSection/CommentSection.tsx [hunk 2]
```diff
@@ -1,16 +1,51 @@
...
   const [showInput, setShowInput] = useState<boolean>(false)
+  const [comment, setComment] = useState<string>()
+
+  const [
+    createApplicationEventMutation,
+    { loading: isCreatingApplicationEvent },
+  ] = useMutation(CreateApplicationEventQuery)
+
+  const saveStaffComment = async (staffComment: string | undefined) => {
+    if (isCreatingApplicationEvent === false && staffComment) {
```
[the remark this answers is not in the document]
Comment (MargretFinnboga) [reply apps/financial-aid/web-veita/src/components/CommentSection/CommentSection.tsx]: Yass good suggestion

File: libs/financial-aid/shared/src/lib/types.ts [hunk 3]
```diff
@@ -22,6 +22,20 @@ export enum ApplicationState {
...
+  DATANEEDED = 'DataNeeded',
+  REJECTED = 'Rejected',
+  APPROVED = 'Approved',
+  STAFFCOMMENT = 'StaffComment',
+}
+// export const ApplicationEventType = { ...ApplicationState, ...EventType }
+// export type ApplicationEventType = typeof ApplicationEventType
+// export type ApplicationEventType = ApplicationState | EventType
+
+// export enum (ApplicationEventType & ApplicationState)
```
Comment (Rikkiabb) [inline libs/financial-aid/shared/src/lib/types.ts:37]: Let's throw this in the trash and forget about it

**Review by Rikkiabb — CHANGES_REQUESTED** [review]

**Review by Rikkiabb — APPROVED** [review]
🐰

**Pull request merged**